portfolio/src/main/webapp/script.js

```diff
@@ -56,3 +56,25 @@ function getRandomName() {
     document.getElementById('name-container').innerText = name;
   });
 }
+
+/**
+ * Fetches names from DataServlet and adds them to the DOM as a list.
+ */
+function getNames() {
+  fetch('/data').then(response => response.json()).then((names) => {
+
+    const namesListElement = document.getElementById('names-container');
+    namesListElement.innerHTML = '';
+    for (const nameIndex in names) {
+      console.log(names[nameIndex]);
```

Comment:
Just a tip: generally at Google we like to avoid checking in `console.log` statements.

They are super-useful when debugging what is going on when you are working on something, but can rapidly stack up if you're not careful and the console is full of noise which can make it harder to debug real issues.

Have you ever used the Chrome/Firefox developer tools?  They let you set breakpoints in the javascript which often means you don't need explicit `console.logs` anyway since you can inspect the the values at a breakpoint anyway - let me know if you'd like me to show you how if you've not seen it before :)

Reply:
Thank you for the tip, I didn't know that's possible, but it is very useful.